src/umpire/strategy/FixedPool.cpp

```diff
@@ -0,0 +1,176 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018-2019, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory
+//
+// Created by David Beckingsale, david@llnl.gov
+// LLNL-CODE-747640
+//
+// All rights reserved.
+//
+// This file is part of Umpire.
+//
+// For details, see https://github.com/LLNL/Umpire
+// Please also see the LICENSE file for MIT license.
+//////////////////////////////////////////////////////////////////////////////
+
+#include "umpire/strategy/FixedPool.hpp"
+
+#include "umpire/util/Macros.hpp"
+
+#include <cstring>
+#include <cstdlib>
+#include <algorithm>
+
+// TODO: Support for Windows
+#include <strings.h>
+
+namespace umpire {
+namespace strategy {
+
+static constexpr size_t bits_per_int = sizeof(int) * 8;
+
+FixedPool::Pool::Pool(AllocationStrategy* allocation_strategy,
+                      const size_t object_bytes, const size_t objects_per_pool,
+                      const size_t avail_bytes) :
+  strategy(allocation_strategy),
+  data(reinterpret_cast<char*>(strategy->allocate(object_bytes * objects_per_pool))),
+  avail(reinterpret_cast<int*>(std::malloc(avail_bytes))),
+  num_avail(objects_per_pool)
+{
+  const unsigned char not_zero = ~0;
```

Comment:
This sets everything to 1? I think anything involving a bit twiddle should have a comment on what is happening and why.